test/models/site_test.rb

```diff
@@ -92,4 +92,35 @@ def test_validate(data)
       assert_equal(expected, site.valid?)
     end
   end
+
+  sub_test_case "Custome style" do
+    test "with #base_hue" do
+      site = create(:site, base_hue: 300)
+
+      assert site.custom_hue_css_url
+    end
+
+    test "without #base_hue" do
+      site = create(:site, base_hue: nil)
+
+      assert_nil site.custom_hue_css_url
+    end
+
+    test "update #base_hue" do
+      site = create(:site, base_hue: nil)
+
+      assert_nil site.custom_hue_css_url
+      site.update!(base_hue: 200)
+      assert_valid_custom_css site
+      site.update!(base_hue: 300)
+      assert_valid_custom_css site
+      site.update!(base_hue: nil)
+      assert_nil site.custom_hue_css_url
+    end
+
+    def assert_valid_custom_css(site)
+      assert_match(/\.css\z/, site.custom_hue_css_url)
+      assert_match(/background-color:/, site.custom_hue_css.read)
```

Comment:
When it failed, it is too hard to correct css content or assertion.
ref: http://www.clear-code.com/blog/2012/1/18.html